utils/writers.ts

```diff
@@ -0,0 +1,66 @@
+import { pipe } from "fp-ts/lib/function";
+import * as TE from "fp-ts/lib/TaskEither";
+import * as E from "fp-ts/lib/Either";
+import { upsertBlobFromObject } from "@pagopa/io-functions-commons/dist/src/utils/azure_storage";
+import { BlobService } from "azure-storage";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import {
+  AssertionFileName,
+  LolliPOPKeysModel,
+  NewLolliPopPubKeys,
+  RetrievedLolliPopPubKeys
+} from "../model/lollipop_keys";
+import { cosmosErrorsToString, ErrorKind, InternalError } from "./errors";
+
+export type PopDocumentWriter = (
+  item: NewLolliPopPubKeys
+) => TE.TaskEither<InternalError, RetrievedLolliPopPubKeys>;
+
+export type AssertionWriter = (
+  assertionFileName: AssertionFileName,
+  assertion: string
+) => TE.TaskEither<InternalError, true>;
+
+// IMPLEMENTATION
+export const getPopDocumentWriter = (
+  lollipopKeysModel: LolliPOPKeysModel
+): PopDocumentWriter => (item): ReturnType<PopDocumentWriter> =>
+  pipe(
+    lollipopKeysModel.upsert(item),
+    TE.mapLeft(error => ({
+      detail: cosmosErrorsToString(error),
+      kind: ErrorKind.Internal as const
+    }))
+  );
+
+export const getAssertionWriter = (
+  assertionBlobService: BlobService,
+  lollipopAssertionStorageContainerName: NonEmptyString
+): AssertionWriter => (
+  assertionFileName,
+  assertion
+): ReturnType<AssertionWriter> =>
+  pipe(
+    TE.tryCatch(
+      () =>
+        upsertBlobFromObject(
```

Comment:
verify if the blob exists before inserting the assertion.
This could also be done using the create instead of upsert.